crates/ide-assists/src/handlers/extract_function.rs

```diff
@@ -1341,6 +1356,52 @@ fn format_function(
     fn_def
 }
 
+fn format_generic_params_and_where_clause(
+    fun: &Function,
+    db: &dyn HirDatabase,
+) -> (String, String) {
+    (format_generic_param_list(fun, db), format_where_clause(fun))
+}
+
+fn format_generic_param_list(fun: &Function, db: &dyn HirDatabase) -> String {
+    let param_type_names: Vec<String> =
+        fun.params.iter().map(|p| p.ty.display(db).to_string()).collect();
```

Comment:
in this first iteration, I'm comparing the parameter `ty` display names to the `GenericParam::TypeParam` names, both as strings. 

This "works" for my test case, but my hunch is that it isn't the "right" way of doing this comparison.

Reply:
I've updated this now to compare the `TypeParam`s more directly, thanks to some advice on zulip from @flodiebold .